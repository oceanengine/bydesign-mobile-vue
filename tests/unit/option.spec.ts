import { mount } from '../index';
import { Checker as CheckerNative, CheckerItem as CheckerItemNative } from '@src';
const Checker = CheckerNative as any;
const CheckerItem = CheckerItemNative as any;

describe('Checker', () => {
  test('render base', () => {
    const wrapper = mount({
      template: `
        <byted-checker v-model="checker">
          <byted-row gutter="10">
            <byted-col span="8">
              <byted-checker-item label="0" :value="0" />
            </byted-col>
            <byted-col span="8">
              <byted-checker-item label="1" value="1" />
            </byted-col>
            <byted-col span="8">
              <byted-checker-item value="2">2</byted-checker-item>
            </byted-col>
          </byted-row>
        </byted-checker>
      `,
      components: {
        'byted-checker': Checker,
        'byted-checker-item': CheckerItem,
      },
      data() {
        return {
          checker: [0, '2'],
        };
      },
    });
    expect(wrapper).toMatchSnapshot();
  });
  test('inputType radio', () => {
    const wrapper = mount({
      template: `
        <byted-checker v-model="checker">
          <byted-checker-item label="0" :value="0" />
          <byted-checker-item label="1" value="1" />
          <byted-checker-item value="2">2</byted-checker-item>
        </byted-checker>
      `,

      components: {
        'byted-checker': Checker,
        'byted-checker-item': CheckerItem,
      },
    });
    expect(wrapper).toMatchSnapshot();
  });

  test('Event triggered when the binding value changes', async () => {
    const onClick = jest.fn();
    const onChange = jest.fn();
    const wrapper = mount({
      template: `
        <byted-checker 
          v-model="checker"
          @click="onClick"
          @change="onChange">
          <byted-checker-item label="0" :value="0" />
          <byted-checker-item label="1" value="1" />
          <byted-checker-item value="2">2</byted-checker-item>
        </byted-checker>
      `,
      data() {
        return {
          checker: ['2'],
        };
      },
      methods: {
        onClick,
        onChange,
      },
      components: {
        'byted-checker': Checker,
        'byted-checker-item': CheckerItem,
      },
    });
    const items = wrapper.findAll('.byted-checker-item');
    await items.at(0).trigger('click');

    expect(onClick).toBeCalledTimes(1);
    expect(onChange).toBeCalledTimes(1);
  });
});
