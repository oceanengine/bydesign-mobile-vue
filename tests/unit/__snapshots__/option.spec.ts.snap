// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checker inputType radio 1`] = `
<div class="byted-checker"><label class="byted-checker-item byted-checker-item--default byted-checker-item--normal"><input type="checkbox" name="" value="0">
    <div class="byted-checker-item__label">0</div>
  </label> <label class="byted-checker-item byted-checker-item--default byted-checker-item--normal"><input type="checkbox" name="" value="1">
    <div class="byted-checker-item__label">1</div>
  </label> <label class="byted-checker-item byted-checker-item--default byted-checker-item--normal"><input type="checkbox" name="" value="2">2</label></div>
`;

exports[`Checker render base 1`] = `
<div class="byted-checker">
  <div class="byted-row" style="margin-left: -5px; margin-right: -5px;">
    <div class="byted-col byted-col--8" style="padding-left: 5px; padding-right: 5px;"><label class="byted-checker-item byted-checker-item--default byted-checker-item--normal byted-checker-item--active"><input type="checkbox" name="" value="0">
        <div class="byted-checker-item__label">0</div><i class="byted-checker-item__icon" style="width: .8rem; height: .8rem;">test-file-stub</i>
      </label></div>
    <div class="byted-col byted-col--8" style="padding-left: 5px; padding-right: 5px;"><label class="byted-checker-item byted-checker-item--default byted-checker-item--normal"><input type="checkbox" name="" value="1">
        <div class="byted-checker-item__label">1</div>
      </label></div>
    <div class="byted-col byted-col--8" style="padding-left: 5px; padding-right: 5px;"><label class="byted-checker-item byted-checker-item--default byted-checker-item--normal byted-checker-item--active"><input type="checkbox" name="" value="2">2<i class="byted-checker-item__icon" style="width: .8rem; height: .8rem;">test-file-stub</i></label></div>
  </div>
</div>
`;
